braindecode/datasets/base.py

```diff
@@ -160,3 +160,31 @@ def split(self, by=None, property=None, split_ids=None):
         return {str(split_name): BaseConcatDataset(
             [self.datasets[ds_ind] for ds_ind in ds_inds])
             for split_name, ds_inds in split_ids.items()}
+
+    @property
+    def metadata(self):
+        """Concatenate the metadata and description of the wrapped Epochs.
+
+        NOTE: This is implemented as a property to avoid having to keep a very
+              large DataFrame in case the dataset contains very long or many
+              recordings.
+
+        Returns
+        -------
+        pd.DataFrame:
+            DataFrame containing as many rows as there are windows in the
+            BaseConcatDataset, with the metadata and description information
+            for each window.
+        """
+        if not all([isinstance(ds, WindowsDataset) for ds in self.datasets]):
+            raise TypeError('Property metadata can only be computed when all '
+                            'datasets are WindowsDataset.')
+
+        all_dfs = list()
+        for ds in self.datasets:
+            df = ds.windows.metadata
+            for k, v in ds.description.items():
+                df[k] = v
+            all_dfs.append(df)
+
+        return pd.concat(all_dfs)
```

Comment:
I guess this is the critical part of this PR with regards to discussing. I am not sure I have a good mental overview over the dataset part. so metadata here will be concatenated from windows datasets metadata as far as I see. 
Regarding implementation as property, could be fine, I am a bit worried as I don't think we use properties in other places, and it can always cause some confusion if there is non-neglible execution time on access to a member (meaning looks like access to a member from caller side). But as said, could be fine.

@gemeinl do you think this metadata here fits in nicely with rest of data structures as you see them? or do you envision any potential conflicts/confusions?

Reply:
I agree with you, this part requires some discussion. I guess this could instead be a method `get_metadata()`, which would make the fact that this requires computations more explicit.

Reply:
@gemeinl can maybe also give is view tomorrow on metadata vs get_metadata() tomorrow. I am not sure. metadata more consistent with windows.metadata, get_metadata() more explicit that there is a computation.... what does yoda @agramfort think?